src/Aspire.Hosting.Azure/api/Aspire.Hosting.Azure.cs

```diff
@@ -205,6 +223,35 @@ public BicepTemplateFile(string path, bool deleteFileOnDispose) { }
         public readonly void Dispose() { }
     }
 
+    public sealed partial class ExistingAzureResourceAnnotation : ApplicationModel.IResourceAnnotation
+    {
+        public ExistingAzureResourceAnnotation(object name, object? resourceGroup = null) { }
+
+        public object Name { get { throw null; } }
+
+        public object? ResourceGroup { get { throw null; } }
+    }
+
+    public static partial class ExistingAzureResourceExtensions
```

Comment:
@captainsafia I was just playing with these and the extension methods should be in the Aspire.Hosting namespace.